File: csharp/lesson 1/BinarySearch.cs
// https://leetcode.com/problems/binary-search

public class Solution
{
    public int BinarySearch(int[] nums, int target) 
    {
        var lo = 0;
        var hi = nums.Length - 1;

        while (lo <= hi)
        {
            var mid = lo + (hi - lo) / 2;
            if (nums[mid] > target) 
            {
                hi = mid - 1;
            }
            else if (nums[mid] < target) 
            {
                lo = mid + 1;
            }
            else 
            {
                return mid;
            }
        }
        
        return -1;
    }
}

File: python/lesson 1/binary_search.py
from typing import List

# https://leetcode.com/problems/binary-search
class Solution:
    def binary_search(self, nums: List[int], target: int) -> int:
        lo, hi = 0, len(nums) - 1
        while lo <= hi:
            mid = lo + (hi - lo) // 2
            if target < nums[mid]:
                hi = mid - 1
            elif target > nums[mid]:
                lo = mid + 1
            else:
                return mid
        return -1
